src/app/dev/DevToys.Tools/Helpers/XmlHelper.cs

```diff
@@ -0,0 +1,129 @@
+﻿using System.Text;
+using System.Text.RegularExpressions;
+using DevToys.Tools.Models;
+using System.Xml;
+using Microsoft.Extensions.Logging;
+using DevToys.Api.Core;
+
+namespace DevToys.Tools.Helpers;
+
+internal static class XmlHelper
+{
+    /// <summary>
+    /// Detects whether the given string is a valid Xml or not.
+    /// </summary>
+    internal static bool IsValid(string? input, ILogger logger)
+    {
+        if (string.IsNullOrWhiteSpace(input))
+        {
+            return false;
+        }
+
+        input = input!.Trim();
```

Comment:
This method may be used a lot due to Smart Detection. `Trim` will allocate another string if there's a space at the beginning or the end. I wonder if we could avoid it by reading the string character by character to find the `<` and `>` without allocating a new string.

Reply:
Updated to go character by character instead of using Trim.